File: multi-db/src/main/java/com/multidb/enums/Gender.java
package com.multidb.enums;

import java.util.Arrays;

public enum Gender {
    M( ""),
    F( "");

    private final String detail;

    Gender(String detail) {
        this.detail = detail;
    }

    public static Gender getEnum(String key) {
        return Arrays.stream(values())
                .filter(s -> s.toString().equals(key))
                .findFirst()
                .orElseThrow(() -> new IllegalArgumentException("No matching Gender for [" + key + "]"));
    }

    public String getDetail() {
        return detail;
    }
}

File: mysql/src/main/kotlin/com/mysql/enums/Gender.kt
package com.mysql.enums

import java.lang.IllegalArgumentException

enum class Gender(val detail: String) {
    M(""),
    F("");

    companion object {
        @JvmStatic
        fun fromString(code : String): Gender =
            values().find { value -> value.toString() == code } ?: throw IllegalArgumentException("No matching Gender for [${code}]")
    }
}